src/ragger/backend/speculos.py

```diff
@@ -248,3 +250,20 @@ def navigate_and_compare_until_snap(self,
         img_idx = self.navigate_until_snap(path, test_case_name, start_img_idx, last_img_idx, True)
         # Then compare all snapshots taken to golden images.
         return self._compare_snaps(path, test_case_name, img_idx)
+
+    def navigate_and_compare(self, path: Path, test_case_name: Path,
+                             instructions: List[NavigationInstruction]) -> bool:
+        snaps_tmp_path = self._get_snaps_dir_path(path, test_case_name, False)
+        snaps_tmp_path.mkdir(parents=True, exist_ok=True)
+
+        # First navigate to the last step and take snapshots of every screen in the flow.
+        for idx, instruction in enumerate(instructions):
```

Comment:
The navigate method should rather have a take snap parameter that would be set to true when called from navigate_and_compare rather than have a loop here and pass each instruction to another loop in navigate...

Reply:
Humf, I'm not found of this one. This would need to pass many parameters to the `navigate` method (snaps dir path, test case name, snap index, ...).
I rather go back and allow only one navigation instruction per `navigate` call.
What do you think?

Reply:
Then you might as well make the navigate method private if it's only used for one navigation instruction (the user can directly use "click_right" "click_left" and so on, I think).